lib/src/launchpad.rs

```diff
@@ -238,6 +248,255 @@ impl LaunchpadMk2 {
+        let output_device = midi.device(output_port)
+            .expect("No Launchpad Pro Output found!");
+
+        let input = midi.input_port(input_device, 1024)
+            .expect("Failed to open port");
+        let output = midi.output_port(output_device, 1024)
+            .expect("Failed to open port");
+
+        LaunchpadPro {
+            input_port: input,
+            output_port: output,
+            midi: None,
+        }
+    }
+
+    /// Set all LEDs to the same color
+    pub fn light_all(&mut self, color: Color) {
+        assert_color(color);
+        // F0h 00h 20h 29h 02h 18h 0Eh <Colour> F7h
+        // Message cannot be repeated.
+        self.output_port
+            .write_sysex(0, &[0xF0, 0x00, 0x20, 0x29, 0x02, 0x10, 0x0E, color, 0xF7])
+            .expect("Fail");
+    }
+
+    /// Set a single LED to flash. Uses a smaller header than `flash_led` or
+    /// `flash_leds` with a single item
+    pub fn flash_single(&mut self, led: &ColorLed) {
+        // ch2
+        // (0x91, <btn>, <color>)
+        assert_position_pro(led.position);
+        assert_color(led.color);
+        self.output_port.write_sysex(0,
+                                     &[0xF0,
+                                       0x00,
+                                       0x20,
+                                       0x29,
+                                       0x02,
+                                       0x10,
+                                       0x23,
+                                       led.position,
+                                       led.color,
+                                       0xF7])
+                        .expect("Fail");
+    }
+
+    /// Set a single LED to pulse. Uses a smaller header than `pulse_led` or
+    /// `pulse_leds` with a single item
+    pub fn pulse_single(&mut self, led: &ColorLed) {
+        // ch3
+        // (0x92, <btn>, <color>)
+
+        self.output_port.write_sysex(0,
+                                     &[0xF0,
+                                       0x00,
+                                       0x20,
+                                       0x29,
+                                       0x02,
+                                       0x10,
+                                       0x28,
+                                       led.position,
+                                       led.color,
+                                       0xF7])
+                        .expect("Fail");
+    }
+
+    /// Set a single LED to a palette color. Use `light_single` instead, its faster.
+    pub fn light_led(&mut self, led: &ColorLed) {
+        // F0h 00h 20h 29h 02h 18h 0Ah <LED> <Colour> F7h
+        // Message can be repeated up to 80 times.
+        self.light_leds(&[led])
+    }
+
+    /// Set LEDs to a certain color. Up to 80 LEDs can be set uniquely at once.
+    pub fn light_leds(&mut self, leds: &[&ColorLed]) {
+        assert!(leds.len() <= 80);
+        for led in leds {
```

Comment:
Probably going to update this to grab the LEDs in chunks of up to 97, as the Pro can have the position & color pairs specified up to that many times in a single SysEx message.